### PR TITLE
add some CommandLineParser tests

### PR DESCRIPTION
**Pull request by alalek**

### CODE REVIEW

File: modules/core/test/test_utils.cpp [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+TEST(CommandLineParser, testPositional_noArgs)
+{
+    const char* argv[] = {"<bin>"};
+    const int argc = 1;
+    cv::CommandLineParser parser(argc, argv, keys2);
+    //EXPECT_FALSE(parser.has("arg1"));
+    //EXPECT_FALSE(parser.has("arg2"));
+    //EXPECT_EQ("default1", parser.get<String>("arg1"));
+    EXPECT_EQ("default1", parser.get<String>(0));
+    //??? EXPECT_ANY_THROW(parser.get<String>("arg2"));
```
Comment (paroj) [inline modules/core/test/test_utils.cpp]: it does not throw, but rather sets the error flag for `CommandLineParser::check`

File: modules/core/test/test_utils.cpp [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    EXPECT_FALSE(parser.has("unused"));
+    EXPECT_FALSE(parser.has("q")); // TODO Throw ???
+}
+
+TEST(CommandLineParser, testBoolOption_noArgs)
+{
+    const char* argv[] = {"<bin>"};
+    const int argc = 1;
+    cv::CommandLineParser parser(argc, argv, keys);
+    EXPECT_FALSE(parser.get<bool>("help"));
```
Comment (paroj) [inline modules/core/test/test_utils.cpp]: this one sets the error flag. maybe add `EXPECT_TRUE(parser.check())`

File: modules/core/test/test_utils.cpp [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    const char* argv[] = {"<bin>", "-h=TRUE", "--info=true"};
+    const int argc = 3;
+    cv::CommandLineParser parser(argc, argv, keys);
+    EXPECT_TRUE(parser.has("help"));
+    EXPECT_TRUE(parser.has("h"));
+    EXPECT_TRUE(parser.has("info"));
+    EXPECT_TRUE(parser.has("i"));
+    EXPECT_FALSE(parser.has("n"));
+    EXPECT_FALSE(parser.has("unused"));
+    EXPECT_FALSE(parser.has("q")); // TODO Throw ???
```
Comment (paroj) [inline modules/core/test/test_utils.cpp]: good idea. asking for a undeclared key is likely a programming error. (e.g. typo in key name)

**Comment by vpisarev** [comment]
thanks! this is definitely useful! :+1:

**Pull request merged**